db/data/20200528063206_set_patient_soft_deletion_info.rb

```diff
@@ -1,15 +1,19 @@
 class SetPatientSoftDeletionInfo < ActiveRecord::Migration[5.2]
   def up
-    bot_user = User.find_or_create_by!(full_name: "Data Migration Bot",
-                                       sync_approval_status: "denied",
-                                       sync_approval_status_reason: "Bot user doesn't require sync") { |user|
-      user.device_created_at = Time.current
-      user.device_updated_at = Time.current
-    }
-
-    Patient.with_discarded.discarded.in_batches(of: 1_000) do |batch|
-      batch.update_all(deleted_by_user_id: bot_user.id, deleted_reason: "unknown")
-    end
+    # This migration conflicts with the validations added on User later
+    # (teleconsultation_phone_number). Commenting this out since it's a
+    # one time migration that's already been run on all envs.
+    #
+    # bot_user = User.find_or_create_by!(full_name: "Data Migration Bot",
+    #                                    sync_approval_status: "denied",
+    #                                    sync_approval_status_reason: "Bot user doesn't require sync") { |user|
+    #   user.device_created_at = Time.current
+    #   user.device_updated_at = Time.current
+    # }
+    #
+    # Patient.with_discarded.discarded.in_batches(of: 1_000) do |batch|
+    #   batch.update_all(deleted_by_user_id: bot_user.id, deleted_reason: "unknown")
+    # end
```

Comment:
Could we instead rescue the exception, skip it only when it's problematic?

```ruby
def up
  # try to do the thing
rescue ActiveRecord::RecordInvalid 
  Rails.logger.info "This data migration could not be run. Skipping."
end
```

Reply:
At the point in time this migration is run, this will always fail because the subsequent migration that adds the columns hasn't been added yet.

Reply:
Hmm yeah fair. Yeah this is clearer 👍